src/stp/new_tactics/AvoidBall.cpp

```diff
@@ -2,54 +2,65 @@
 // Created by jesse on 30-04-20.
 //
 
-#include "utilities/GameStateManager.hpp"
 #include "stp/new_tactics/AvoidBall.h"
+
+#include "stp/invariants/game_states/StopGameStateInvariant.h"
 #include "stp/new_skills/GoToPos.h"
+#include "utilities/GameStateManager.hpp"
 #include "world/FieldComputations.h"
 
 namespace rtt::ai::stp::tactic {
 
-    AvoidBall::AvoidBall() {
-        skills = rtt::collections::state_machine<Skill, Status, StpInfo>{skill::GoToPos()};
+AvoidBall::AvoidBall() { skills = rtt::collections::state_machine<Skill, Status, StpInfo>{skill::GoToPos()}; }
+
+void AvoidBall::onInitialize() noexcept {}
+
+void AvoidBall::onUpdate(Status const &status) noexcept {}
+
+void AvoidBall::onTerminate() noexcept {
+    // Call terminate on all skills
+    for (auto &x : skills) {
+        x->terminate();
     }
+}
 
-    void AvoidBall::onInitialize() noexcept {}
+StpInfo AvoidBall::calculateInfoForSkill(StpInfo const &info) noexcept {
+    StpInfo skillStpInfo = info;
 
-    void AvoidBall::onUpdate(Status const &status) noexcept {}
+    // Code for stop state is here
+    if (GameStateManager::getCurrentGameState().getStrategyName() == "stop") {
+        auto ballPosition = info.getBall()->get()->getPos();
+        auto targetPosition = info.getPositionToMoveTo().value();
 
-    void AvoidBall::onTerminate() noexcept {
-        // Call terminate on all skills
-        for (auto &x : skills) {
-            x->terminate();
+        if (targetPosition.dist(ballPosition) < 0.5) {
```

Comment:
Magic value :scream: You should always create a constant variable for these numbers with documentation explaining what the meaning is of this 0.5 value